src/path.rs

```diff
@@ -0,0 +1,790 @@
+/// ```rust,no_run
+/// use tauri_sys::path::app_log_dir;
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_log_dir_path = app_log_dir().await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn app_log_dir() -> crate::Result<PathBuf> {
+    let raw = inner::appLogDir().await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Resolves a sequence of `paths` or `path` segments into an absolute path.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::{resolve, app_data_dir};
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_data_dir_path = app_data_dir().await?;
+///
+/// let path = resolve([app_data_dir_path, "..", "users", "tauri", "avatar.png"]).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn resolve(paths: impl IntoIterator<Item = &str>) -> crate::Result<PathBuf> {
+    let paths = paths.into_iter();
+    let raw = inner::resolve(serde_wasm_bindgen::to_value(&paths.collect::<Vec<_>>())?).await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Normalizes the given `path`, resolving `'..'` and `'.'` segments and resolve symbolic links.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::{normalize, app_data_dir};
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_data_dir_path = app_data_dir().await?;
+///
+/// let path = normalize([app_data_dir_path, "..", "users", "tauri", "avatar.png"]).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn normalize(path: &str) -> crate::Result<PathBuf> {
+    let raw = inner::normalize(JsValue::from_str(path)).await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+///  Joins all given `path` segments together using the platform-specific separator as a delimiter, then normalizes the resulting path.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::{join, app_data_dir};
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_data_dir_path = app_data_dir().await?;
+///
+/// let path = join([app_data_dir_path, "..", "users", "tauri", "avatar.png"]).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn join(paths: impl IntoIterator<Item = &str>) -> crate::Result<PathBuf> {
+    let paths = paths.into_iter();
+    let raw = inner::join(serde_wasm_bindgen::to_value(&paths.collect::<Vec<_>>())?).await?;
+
+    Ok(serde_wasm_bindgen::from_value(raw)?)
+}
+
+/// Returns the directory name of a `path`. Trailing directory separators are ignored.
+///
+/// # Example
+///
+/// ```rust,no_run
+/// use tauri_sys::path::{dirname, app_data_dir};
+///
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let app_data_dir_path = app_data_dir().await?;
+///
+/// let dir = dirname(app_data_dir_path).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[inline(always)]
+pub async fn dirname(path: &str) -> crate::Result<PathBuf> {
```

Comment:
Should this accept `AsRef<Path>` instead of `&str`?

Reply:
Updated in 1fb776d